File: app/src/main/java/me/ykrank/s1next/view/adapter/delegate/NoteAdapterDelegate.java
package me.ykrank.s1next.view.adapter.delegate;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

import java.util.List;

import me.ykrank.s1next.data.api.model.Note;
import me.ykrank.s1next.databinding.ItemNoteBinding;
import me.ykrank.s1next.viewmodel.NoteViewModel;

/**
 * Created by ykrank on 2017/1/5.
 */

public class NoteAdapterDelegate extends BaseAdapterDelegate<Note, NoteAdapterDelegate.BindingViewHolder> {

    public NoteAdapterDelegate(Context context) {
        super(context);
    }

    @NonNull
    @Override
    protected Class<Note> getTClass() {
        return Note.class;
    }

    @Override
    public void onBindViewHolderData(Note note, int position, @NonNull BindingViewHolder holder, @NonNull List<Object> payloads) {
        ItemNoteBinding binding = holder.binding;
        binding.getModel().data.set(note);
        binding.executePendingBindings();
    }

    @NonNull
    @Override
    protected RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent) {
        ItemNoteBinding binding = ItemNoteBinding.inflate(mLayoutInflater, parent, false);
        binding.setModel(new NoteViewModel());
        return new BindingViewHolder(binding);
    }

    static final class BindingViewHolder extends RecyclerView.ViewHolder {

        private final ItemNoteBinding binding;

        public BindingViewHolder(ItemNoteBinding binding) {
            super(binding.getRoot());

            this.binding = binding;
        }
    }
}

File: app/src/main/java/me/ykrank/s1next/view/adapter/delegate/NoteAdapterDelegate.kt
package me.ykrank.s1next.view.adapter.delegate

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.ViewGroup
import com.github.ykrank.androidtools.ui.adapter.simple.SimpleRecycleViewHolder

import me.ykrank.s1next.data.api.model.Note
import me.ykrank.s1next.databinding.ItemNoteBinding
import me.ykrank.s1next.viewmodel.NoteViewModel

/**
 * Created by ykrank on 2017/1/5.
 */

class NoteAdapterDelegate(context: Context) : BaseAdapterDelegate<Note, SimpleRecycleViewHolder<ItemNoteBinding>>(context, Note::class.java) {

    override fun onBindViewHolderData(t: Note, position: Int, holder: SimpleRecycleViewHolder<ItemNoteBinding>, payloads: List<Any>) {
        val binding = holder.binding
        binding.model?.data?.set(t)
    }

    override fun onCreateViewHolder(parent: ViewGroup): RecyclerView.ViewHolder {
        val binding = ItemNoteBinding.inflate(mLayoutInflater, parent, false)
        binding.model = NoteViewModel()
        return SimpleRecycleViewHolder<ItemNoteBinding>(binding)
    }

}
